apollo-federation/src/query_graph/build_query_graph.rs

```diff
@@ -2333,42 +2344,66 @@ fn resolvable_key_applications<'doc>(
 }
 
 fn parse_context(field: &str) -> Result<(String, String), FederationError> {
-    let pattern = Regex::new(
-        r#"^(?:[\n\r\t ,]|#[^\n\r]*)*\$(?:[\n\r\t ,]|#[^\n\r]*)*([A-Za-z_]\w*)([\s\S]*)$"#,
-    )
-    .unwrap();
-
-    let mut iter = pattern.captures_iter(field);
-
-    let Some(captures) = iter.next() else {
-        bail!("Expected to find the name of a context and a selection inside the field argument to `@fromContext`: {field:?}");
-    };
-
-    if iter.next().is_some() {
-        bail!("Expected only one context and selection pair inside the field argument to `@fromContext`: {field:?}");
+    // PORT_NOTE: The original JS regex, as shown below
+    //   /^(?:[\n\r\t ,]|#[^\n\r]*(?![^\n\r]))*\$(?:[\n\r\t ,]|#[^\n\r]*(?![^\n\r]))*([A-Za-z_]\w*(?!\w))([\s\S]*)$/
+    // makes use of negative lookaheads, which aren't supported natively by Rust's regex crate.
+    // There's a fancy_regex crate which does support this, but in the specific case above, the
+    // negative lookaheads are just used to ensure strict *-greediness for the preceding expression
+    // (i.e., it guarantees those *-expressions match greedily and won't backtrack).
+    //
+    // We can emulate that in this case by matching a series of regexes instead of a single regex,
+    // where for each regex, the relevant *-expression doesn't backtrack by virtue of the rest of
+    // the haystack guaranteeing a match. Also note that Rust has (?s:.) to match all characters
+    // including newlines, which we use in place of JS's common regex workaround of [\s\S].
+    fn strip_leading_ignored_tokens(input: &str) -> Result<&str, FederationError> {
+        let leading_pattern = Regex::new(r#"^(?:[\n\r\t ,]|#[^\n\r]*)*((?s:.)*)$"#).unwrap();
```

Comment:
nit: should we make this regex static so we don't need to recreate it?

Reply:
We can make it work with `lazy_static!`, addressed in [d71dff0](https://github.com/apollographql/router/pull/6380/commits/d71dff0f9845031253f11fd2e5aaf6d4d0573279).